### DISCUSSION TITLE
zosmf profile vs rse profile

### DISCUSSION THREAD

**Discussion by kraabrsg** [IBM/zopeneditor-about#453] [Q&A]
Hi we currently use our own vscode plugins for various tasks and to make the development process better suited to our needs, etc.

For cobol we use z open editor and zowe explorer, with a custom vscode plugin. 
For pli/pl1 we use a custom vscode plugins.

The build process is started via REXX (from a tso command).

Now it is sometimes a Problem when the cpu load is high, that zosmf is unable to do its job (i think it was 502-er RC from ZOSMF, cannot connect, saving files is not possible). We are currently in a long Time testing Phase, if we should use VSCODE + plugins as a productive System.

What is the best option using RSE or ZOSMF (in terms of cpu usage, etc..., performance, maintainability, etc ). Setup our own zosmf or configure RSE API, we now use also RDZ / IDZ for our development.... (i read a comment about high zosmf cpu usage in a git issue)

Thanks!

**Comment by phaumer** [comment]
Hello @kraabrsg. As we are the RSE development team we are biased against our own solution. For RSE we would be happy to receive requirements and feedback here or via support tickets that will help us to improve our solution. I am attaching an overview deck that we used for customer presentations comparing the two from our point of view.

[Wazi-VS-Code-zOSMF-RSEAPI.pdf](https://github.com/user-attachments/files/17385243/Wazi-VS-Code-zOSMF-RSEAPI.pdf)

**Reply by kraabrsg** [reply]
@phaumer thanks for the input, i am curious about the license... To use the RSE API you don't a bought license like IDZ (just install and configure IBM Explorer for z/os https://www.ibm.com/docs/en/explorer-for-zos/3.3?topic=documentation-rse-api  ) -> correct ?

**Reply by phaumer** [reply]
Yes, it is available for free without technical support as part of the free z/OS Explorer. It is also part of IDz and utilized for many of its integrations. It is also part of IDzEE, which also includes technical support for Z Open Editor and its RSE API use case; plus additional features shown on the slides such as the resources table with search.
